piccolo_admin/example.py

```diff
@@ -442,6 +443,22 @@ def booking_endpoint(request: Request, data: BookingModel) -> str:
     ],
     auth_table=User,
     session_table=Sessions,
+    charts=[
+        ChartConfig(
+            title="Movie count",
+            chart_type="Pie",
+            data=[
+                ["George Lucas", 4],
+                ["Peter Jackson", 6],
+                ["Ron Howard", 1],
+            ],
+        ),
+        ChartConfig(
+            title="Director gender",
+            chart_type="Column",
+            data=[["Male", 7], ["Female", 3]],
+        ),
+    ],
```

Comment:
I'd add examples for each chart we support - I think we support Pie, Line, Column, Bar and Area.

Also, I think the main missing piece if the data is hardcoded at the moment, but in the docs we show how the user can pass in an async function, so the data can be fetched dynamically.

Reply:
Yes, you're right, the data is hardcoded because we can't run database queries in the piccolo admin example (`piccolo_conf` error), but we can use a coroutine or sync function in a normal application. I will change the test to run the hardcoded data from the coroutine in `example.py` so that the user can see an example of how to use it. Something like this
```python
async def fetch_chart_data() -> t.List[t.List[t.Any]]:
    return [
        ["George Lucas", 4],
        ["Peter Jackson", 6],
        ["Ron Howard", 1],
    ]

chart_data = asyncio.run(fetch_chart_data())

APP = create_admin(charts=[
        ChartConfig(
            title="Movie count Pie",
            chart_type="Pie",
            data=chart_data,
        ),
        ChartConfig(
            title="Movie count Line",
            chart_type="Line",
            data=chart_data,
        ),
        ChartConfig(
            title="Movie count Column",
            chart_type="Column",
            data=chart_data,
        ),
        ChartConfig(
            title="Movie count Bar",
            chart_type="Bar",
            data=chart_data,
        ),
        ChartConfig(
            title="Movie count Area",
            chart_type="Area",
            data=chart_data,
        ),
  ],
)
```

Reply:
I think the problem is the query is being run before the engine has been assigned to the tables.

If we make it dynamic then we won't have this problem.

```python
async def fetch_chart_data() -> t.List[t.List[t.Any]]:
    return await Movie.select() ...

APP = create_admin(charts=[
        ChartConfig(
            title="Movie count Pie",
            chart_type="Pie",
            source=fetch_chart_data,
        ),
]
```

Then every time the chart is fetched from the API, we run the function to get the latest data.

Reply:
I can make the change if you like - it's up to you.

Reply:
Of course. That would be great. In the last commit I made your suggested changes. I think I covered everything except one thing I had trouble wrapping the chart pages in the `BaseView` (to have a sidebar) due to a weird Vue router warning 
`vue-router.esm.js:16 [vue-router] missing param for named route "chart": Expected "chartSlug" to be defined` even though the `chartSlug` is defined in the router like this
```javasript
{
    path: '/charts/:chartSlug',
    name: 'chart',
    component: Chart,
    props: true
},
```
so I left the `DetailBaseView`. It would be nice if you could look at that too. Thanks

Reply:
OK, I'll create a branch, and will play around.